modules/nix/default.nix

```diff
@@ -836,18 +838,18 @@ in
     users.users = mkIf configureBuildUsers nixbldUsers;
 
     # Not in NixOS module
-    users.groups.nixbld = mkIf configureBuildUsers {
+    users.groups.nixbld = {
       description = "Nix build group for nix-daemon";
       gid = config.ids.gids.nixbld;
       members = attrNames nixbldUsers;
```

Comment:
This should probably be empty if `!configureBuildUsers`.

Reply:
But then the group wouldn't be created if `auto-allocate-uids` is true? I mean, to be fair, I haven't seen it actually documented anywhere that the group is required. Maybe I'm wrong or my config is messed up somehow. But both on my NixOS and Darwin machines, if the group doesn't exist, it just gives the error:

```
error: the group 'nixbld' specified in 'build-users-group' does not exist
```

If the group is created, even though it has no members, it will work fine and create the UIDs. If the `build-users-group` is set to be empty, it will just build everything as root.

Reply:
I just mean `members`; the group would still be created. Right now `nixbldUsers` will still contain a bunch of users even if `auto-allocate-uids` is on, and they’ll be listed as group members here, but the users won’t necessarily exist. That seems bad. https://github.com/LnL7/nix-darwin/pull/1335#discussion_r1954705705 would fix this by making `nixbldUsers` empty in that case, but it’d be more work than just using `lib.optionals` here.

Reply:
Fixed by implementing https://github.com/LnL7/nix-darwin/pull/1335#discussion_r1954705705, now the `_nixbld` users 1-128 are always considered "known".